File: csharp/TextManagerHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

namespace TextObfuscation
{
    class TextManagerHelper
    {
        public IEnumerable<TextPart> Convert(string text)
        {
            var contents = "";
            var type = TextPartType.WORD;
            for (var i = 0; i < text.Length; i++)
            {
                if (char.IsLetter(text[i]))
                {
                    if (type == TextPartType.WORD)
                    {
                        contents += text[i];
                    }
                    else
                    {
                        if (contents != "") yield return new TextPart(contents, type);
                        contents = text[i].ToString();
                        type = TextPartType.WORD;
                    }
                }
                else
                {
                    if (type == TextPartType.NONWORD)
                    {
                        contents += text[i];
                    }
                    else
                    {
                        if (contents != "") yield return new TextPart(contents, type);
                        contents = text[i].ToString();
                        type = TextPartType.NONWORD;
                    }
                }
            }
            yield return new TextPart(contents, type);
        }
    }
}

File: java/src/com/company/TextManagerHelper.java
package com.company;

import java.util.ArrayList;
import java.util.List;

class TextManagerHelper {

    public List<TextPart> convert(String text) {
        List<TextPart> parts = new ArrayList<TextPart>();
        String contents = "";
        TextPartType type = TextPartType.WORD;
        for (var i = 0; (i < text.length()); i++) {
            if (Character.isLetter(text.charAt(i))) {
                if ((type == TextPartType.WORD)) {
                    contents += text.charAt(i);
                }
                else {
                    if ((contents != "")) {
                        parts.add(new TextPart(contents, type));
                    }
                    contents = Character.toString(text.charAt(i));
                    type = TextPartType.WORD;
                }

            }
            else if ((type == TextPartType.NONWORD)) {
                contents += Character.toString(text.charAt(i));
            }
            else {
                if ((contents != "")) {
                    parts.add(new TextPart(contents, type));
                }
                contents = Character.toString(text.charAt(i));
                type = TextPartType.NONWORD;
            }

        }
        return parts;
    }
}